src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -0,0 +1,522 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.programgroups.StructuralVariationSparkProgramGroup;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import scala.Tuple2;
+import scala.Tuple3;
+
+import java.io.*;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+// TODO: choose which parameters allowed to be tunable
+// TODO: if throws, would temp files be cleaned up automatically?
+// TODO: most robust way to check OS in method runSGAModulesInSerial (Mac doesn't support multi-thread mechanism in SGA)?
+@CommandLineProgramProperties(
+        summary        = "Program to call SGA to perform local assembly and return assembled contigs.",
+        oneLineSummary = "Perform SGA-based local assembly on fasta files on Spark",
+        programGroup   = StructuralVariationSparkProgramGroup.class)
+public final class RunSGAViaProcessBuilderOnSpark extends GATKSparkTool {
+    private static final long serialVersionUID = 1L;
+
+    @Argument(doc       = "Absolute path to SGA installation.",
+              shortName = "sgaPath",
+              fullName  = "fullPathToSGA",
+              optional  = false)
+    public String pathToSGA = null;
+
+    @Argument(doc       = "An URI to headerless file where each line contains the absolute path to one raw FASTQ file " +
```

Comment:
I'd describe the input format as, "each line contains a breakpoint ID and the absolute path to one raw FASTQ file delimited by a tab"